src/expression/nn.cpp

```diff
@@ -45,3 +45,43 @@ deepworks::Softmax::Softmax(std::string name)
 deepworks::Shape deepworks::Softmax::outShape(const deepworks::Shape& in_shape) {
     return in_shape;
 }
+
+deepworks::BatchNorm::BatchNorm(float eps, float alpha, std::string name)
+    : BaseOp<deepworks::BatchNorm>(LayerInfo(std::move(name), "BatchNorm1D")) {
+    m_info.impl().attrs["eps"] = eps;
+    m_info.impl().attrs["alpha"] = alpha;
+}
+
+void deepworks::BatchNorm::init(const Shape& in_shape) {
+
+    // NB: Init gamma
+    deepworks::Tensor gamma(deepworks::Shape{in_shape[1]});
+    deepworks::initializer::constant(gamma, 0.2);
+    m_info.impl().params.emplace_back(std::move(gamma), true);
+
+    // NB: Init beta.
+    deepworks::Tensor beta(deepworks::Shape{in_shape[1]});
+//    deepworks::initializer::zeros(beta);
+    deepworks::initializer::constant(beta, 0.1);
+    m_info.impl().params.emplace_back(std::move(beta), true);
+
+    deepworks::Tensor input_centered(in_shape);
+    deepworks::initializer::zeros(input_centered);
```

Comment:
I believe `input_centered`, `std`, `running_mean`, `running_var` - are't `Paramaters` at all, because the aren't trainable, it's just a temporary buffers.

So it can be placed in `CPUBactchnorm1D`

Reply:
@l-bat Correct me

Reply:
Done